libnmstate/nmstate.py

```diff
@@ -166,17 +167,47 @@ def _get_interface_info_from_plugins(plugins):
         for iface in plugin.get_interfaces():
             iface[IFACE_PRIORITY_METADATA] = plugin.priority
             iface_name = iface[Interface.NAME]
-            if iface_name in all_ifaces:
-                existing_iface = all_ifaces[iface_name]
+            iface_type = iface.get(Interface.TYPE, InterfaceType.UNKNOWN)
+            iface_index = f"{iface_type}.{iface_name}"
+            if iface_index in all_ifaces:
+                existing_iface = all_ifaces[iface_index]
                 existing_priority = existing_iface[IFACE_PRIORITY_METADATA]
                 current_priority = plugin.priority
                 if current_priority > existing_priority:
                     merge_dict(iface, existing_iface)
-                    all_ifaces[iface_name] = iface
+                    all_ifaces[iface_index] = iface
                 else:
                     merge_dict(existing_iface, iface)
             else:
-                all_ifaces[iface_name] = iface
+                all_ifaces[iface_index] = iface
+
+    # For any unknown interface, we merge it when there is only one interface
+    # with the same name and valid type
+    to_be_removed_index = []
+    for iface in all_ifaces.values():
+        iface_type = iface.get(Interface.TYPE, InterfaceType.UNKNOWN)
+        iface_name = iface[Interface.NAME]
+        if iface_type == InterfaceType.UNKNOWN:
+            iface_types = _get_iface_types_by_name(
+                all_ifaces.values(), iface_name
+            )
+            if len(iface_types) == 1:
```

Comment:
What if there are 2 such types or even more? It may be safer to use `> 0`, no?

Reply:
As it state in the comment. We only merge unknown interface data to valid interface when there is only __single__ valid interface available.